common/os.cpp

```diff
@@ -261,12 +261,28 @@ Some auxillary functionalities might be affected. Please report this message if
         else
         {
             GUID folder;
+            HRESULT hr;
             switch (f)
             {
             case user_desktop: folder = FOLDERID_Desktop;
                 break;
             case user_documents: folder = FOLDERID_Documents;
-                break;
+                // Use for documents with SHGetFolderPath instead of SHGetKnownFolderPath because SHGetKnownFolderPath 
+                // Doesn't give the path to intel one drive documents folder but to the local documents folder
```

Comment:
Don't mention Intel

Reply:
done

Reply:
Need a more complete description of the issue:

```
// The user's Documents folder location may get overridden, as we know OneDrive does in certain circumstances.
// In such cases, the new function, SHGetKnownFolderPath, does not always return the new path, while the deprecated
// function does.
```